pkg/checkpoint/config_map.go

```diff
@@ -50,13 +52,70 @@ func (c *checkpointer) checkpointConfigMap(namespace, podName, configMapName str
 
 	// TODO(aaron): No need to store if already exists
 	for f, d := range configMap.Data {
-		if err := writeAndAtomicRename(filepath.Join(basePath, f), []byte(d), uid, gid, 0600); err != nil {
+		if err := writeAndAtomicRename(filepath.Join(basePath, f), []byte(d), uid, gid, defaultConfigMapMode); err != nil {
 			return "", fmt.Errorf("failed to write configMap %s: %v", configMap.Name, err)
 		}
 	}
 	return basePath, nil
 }
 
+// checkpointConfigMapProjection will locally store configMap data from volumes with ProjectedVolumeSources.
+// The path to the configmap data becomes: checkpointConfigMapPath/namespace/podname/volumename/configmap.file
+// Where each "configmap.file" is a path from the ConfigMapProjection.KeyToPath field.
+func (c *checkpointer) checkpointConfigMapProjection(namespace, podName, volumeName string, configMapProjection *v1.ConfigMapProjection, uid, gid int) error {
+
+	configMapName := configMapProjection.Name
+	isOptional := configMapProjection.Optional != nil && *configMapProjection.Optional
+
+	configMap, err := c.apiserver.Core().ConfigMaps(namespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil && isOptional {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to retrieve configMap %s/%s: %v", namespace, configMapName, err)
+	}
+
+	basePath := configMapPath(namespace, podName, volumeName)
+	if err = os.MkdirAll(basePath, 0700); err != nil {
+		return fmt.Errorf("failed to create configMap checkpoint path %s: %v", basePath, err)
+	}
+	if err = os.Chown(basePath, uid, gid); err != nil {
```

Comment:
nit: We're potentially creating parent directories via `MkdirAll` call above, but this would only set ownership on the full path.